app/src/main/java/ch/epfl/sweng/swenggolf/database/FireDatabase.java

```diff
@@ -23,10 +26,12 @@
  */
 public final class FireDatabase extends Database {
```

Comment:
I don't understand, why does `FakeDatabase`extends `DatabaseListHandler`but `FireDatabase` still extends only `Database`? `FireDatabase`also needs to handle lists? (I am the one confused here, your code is probably correct)

Reply:
Since codeclimate was complaining about the fact that there were too much lines or methods, I just extracted the code. However since the list management on firebase is done directly with firebase using queries (and not on the lists themselves) it's a bit harder to extract the code. I didn't want to waste to much time on this so I just added a layer of inheritance. If you have a better idea using containment it might be nice.

Reply:
And the problem also resides in the fact that those methods are methods of Database and thus need to be implemented. I removed the public modifier of the class. So only classes of the package can see it.